v0.12.0 Release notes (#1216)
spiceai/spiceai: docs/release_notes/v0.12.0-alpha.md

# Spice v0.12.0-alpha (Apr 29, 2024)

The v0.11.1-alpha release ...

## Highlights

...

## Contributors

- @phillipleblanc
- @Jeadie
- @ewgenius
- @sgrebnov
- @y-f-u
- @lukekim
- @digadeesh
- @Sevenannn

> - @gloomweaver

Reviewer: @ewgenius make sure to update this list and call out david's first contribution.

Author: Updated

Reviewer 2: This needs to be updated to include #1204

## New in this release

- **Clickhouse data connector**: Use Clickhouse as a data source.

- **Spice Client SDK Support**: TBD

> - **Reloading accelerated datasets**: TBD

Reviewer: @ewgenius not a major feature.

Author: Yes, removed

> # Spice v0.12.0-alpha (Apr 29, 2024)

Reviewer: ```suggestion
# Spice v0.12-alpha (Apr 29, 2024)
```

> The 0.12.0-alpha release introduces Clickhouse and Spark data connectors and support for temporal refresh data periods. In addition, official Spice Client SDKs released with support for Spice OSS, and overall stability improvements have been added.

Reviewer: ```suggestion
The v0.12-alpha release introduces Clickhouse and Apache Spark data connectors, adds support for limiting refresh periods for temporal datasets, and includes upgraded Spice Client SDKs compatible with Spice OSS.
```

## Highlights

> - **Clickhouse data connector**: Use Clickhouse databases as a data source.

Reviewer: ```suggestion
- **Clickhouse data connector**: Use [Clickhouse](https://clickhouse.com) as a data source with the `clickhouse:` scheme.
```

> - **Spark data connector**: Use Apache Spark connections as a data source.

Reviewer: ```suggestion
- **Apache Spark Connect data connector**: Use Apache [Spark Connect](https://spark.apache.org/spark-connect/) connections as a data source using the `spark:` scheme.
```

- **Temporal data refresh period**: Set specific time windows for accelerated datasets refresh, ensuring only the most recent data retained.

- **Optional ODBC data connector**: Use ODBC connections as a data source. This feature can be enabled by using the `odbc` cargo feature when building from the trunk.

- **Spice Client SDK Support**: Official Spice.ai SDKs have been released with support for Spice OSS.
  - [spice-rs v2.0.0](https://github.com/spiceai/spice-rs/releases/tag/v2.0.0)
  - [spice.js v2.0.0](https://github.com/spiceai/spice.js/releases/tag/v2.0.0)
  - [gospice v6.0.0](https://github.com/spiceai/gospice/releases/tag/v6.0.0)
  - [spicepy v2.0.0](https://github.com/spiceai/spicepy/releases/tag/v2.0.0)

## Contributors

- @phillipleblanc
- @Jeadie
- @ewgenius
- @sgrebnov
- @y-f-u
- @lukekim
- @digadeesh
- @gloomweaver

> - @edmondop

Reviewer: Should add mach-kernel to this list too.

Author: Added

# Spice v0.12-alpha (Apr 29, 2024)

> The v0.12-alpha release introduces Clickhouse and Apache Spark data connectors, adds support for limiting refresh periods for temporal datasets, and includes upgraded Spice Client SDKs compatible with Spice OSS.

Reviewer 2: This also needs a breaking change notice for `refresh_interval` -> `refresh_check_interval`

Reviewer: I added this below.

> The v0.12-alpha release introduces Clickhouse and Apache Spark data connectors, adds support for limiting refresh periods for temporal datasets, and includes upgraded Spice Client SDKs compatible with Spice OSS.

Reviewer: ```suggestion
The v0.12-alpha release introduces Clickhouse and Apache Spark data connectors, adds support for limiting refresh data periods for temporal datasets, and includes upgraded Spice Client SDKs compatible with Spice OSS.
```

## Highlights

- **Clickhouse data connector**: Use [Clickhouse](https://clickhouse.com) as a data source with the `clickhouse:` scheme.

- **Apache Spark Connect data connector**: Use Apache [Spark Connect](https://spark.apache.org/spark-connect/) connections as a data source using the `spark:` scheme.

> - **Temporal data refresh period**: Set specific time windows for accelerated datasets refresh, ensuring only the most recent data retained.

Reviewer: ```suggestion
- **Refresh data window**: Limit accelerated dataset data refreshes to the specified window,  using the `refresh_data_window` configurations setting as a duration from now configuration setting, for faster and more efficient data refreshes.
```

> - **Optional ODBC data connector**: Use ODBC connections as a data source. This feature can be enabled by using the `odbc` cargo feature when building from the trunk.

Reviewer: ```suggestion
- **ODBC data connector**: Use ODBC connections as a data source using the `odbc:` scheme. The ODBC data connector is currently optional and not included in default builds. It can be conditionally compiled using the `odbc` cargo feature when building from the trunk.
```

> - **ODBC data connector**: Use ODBC connections as a data source using the `odbc:` scheme. The ODBC data connector is currently optional and not included in default builds. It can be conditionally compiled using the `odbc` cargo feature when building from the trunk.

Reviewer 2: ```suggestion
- **ODBC data connector**: Use ODBC connections as a data source using the `odbc:` scheme. The ODBC data connector is currently optional and not included in default builds. It can be conditionally compiled using the `odbc` cargo feature when building from source.
```

> - **Spice Client SDK Support**: Official Spice.ai SDKs have been released with support for Spice OSS.

Reviewer: ```suggestion
- **Spice Client SDK Support**: The official Spice SDKs have been upgraded with support for Spice OSS.
```

  - [spice-rs v2.0.0](https://github.com/spiceai/spice-rs/releases/tag/v2.0.0)
  - [spice.js v2.0.0](https://github.com/spiceai/spice.js/releases/tag/v2.0.0)
  - [gospice v6.0.0](https://github.com/spiceai/gospice/releases/tag/v6.0.0)
  - [spicepy v2.0.0](https://github.com/spiceai/spicepy/releases/tag/v2.0.0)

> 

Reviewer: ```suggestion
## Breaking Changes
- **Refresh interval:** The `refresh_interval` acceleration setting and been changed to `refresh_check_interval` to make it clearer it is the check versus the data interval.
```

## Contributors

- @phillipleblanc
- @Jeadie
- @ewgenius
- @sgrebnov
- @y-f-u
- @lukekim
- @digadeesh
- @gloomweaver
- @edmondop
- @mach-kernel

## New Contributors

> * @mach-kernel made their first contribution in https://github.com/spiceai/spiceai/pull/1204

Reviewer: ```suggestion
* Thanks to @mach-kernel who made their first contribution in https://github.com/spiceai/spiceai/pull/1204 by adding the ODBC data connector!
```

- **Refresh data window**: Limit accelerated dataset data refreshes to the specified window, as a duration from now configuration setting, for faster and more efficient refreshes.

- **ODBC data connector**: Use ODBC connections as a data source using the `odbc:` scheme. The ODBC data connector is currently optional and not included in default builds. It can be conditionally compiled using the `odbc` cargo feature when building from source.

- **Spice Client SDK Support**: The official Spice SDKs have been upgraded with support for Spice OSS.
  - [spice-rs v2.0.0](https://github.com/spiceai/spice-rs/releases/tag/v2.0.0)
  - [spice.js v2.0.0](https://github.com/spiceai/spice.js/releases/tag/v2.0.0)
  - [gospice v6.0.0](https://github.com/spiceai/gospice/releases/tag/v6.0.0)
  - [spicepy v2.0.0](https://github.com/spiceai/spicepy/releases/tag/v2.0.0)
## Breaking Changes
- **Refresh interval:** The `refresh_interval` acceleration setting and been changed to `refresh_check_interval` to make it clearer it is the check versus the data interval.

> ## Contributors

Reviewer: ```suggestion
  - [spicepy v2.0.0](https://github.com/spiceai/spicepy/releases/tag/v2.0.0)
  
## Breaking Changes
- **Refresh interval:** The `refresh_interval` acceleration setting and been changed to `refresh_check_interval` to make it clearer it is the check versus the data interval.

## Contributors
```
